subsys/bluetooth/controller/hci/hci.c

```diff
@@ -4798,6 +4798,138 @@ static void vs_read_tx_power_level(struct net_buf *buf, struct net_buf **evt)
 	rp->handle = sys_cpu_to_le16(handle);
 }
 #endif /* CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL */
+
+#if defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
+/* A memory pool for vandor specific events for fatal error rerporting purposes. */
+NET_BUF_POOL_FIXED_DEFINE(vs_err_tx_pool, 1, BT_BUF_RX_SIZE, 8, NULL);
+
+/* The alias to for convenience of Controller HCI implementation. Controller is build for
+ * a particular architecture hence the alias will allow to avoid conditional compilation.
+ * Host may be not aware of hardware architecture the Controller is working on, hence
+ * all CPU data types for supported architectures should be available during build.
+ */
+#if defined(CONFIG_CPU_CORTEX_M)
+typedef struct bt_hci_vs_fata_error_cpu_data_cortex_m bt_hci_vs_fatal_error_cpu_data;
+
+static void vs_err_fatal_cpu_data_fill(bt_hci_vs_fatal_error_cpu_data *cpu_data,
```

Comment:
Hmm, is `bt_hci_vs_fatal_error_cpu_data` a struct? Looks like the name of a variable instead.

Reply:
It is actually a `typedef`. The name was chosen to follow convention that I found for vendor specific types/structures in source code. Here are some examples: `bt_hci_cp_vs_read_tx_power_level`, `bt_hci_cp_vs_set_min_num_used_chans`.

Have you got a suggestion how to name it?

Reply:
Oh sorry, I had simply mistaken the typedef for a variable >.> 

You can ignore my comment :)